media/js/base/mozilla-utils.js

```diff
@@ -42,12 +42,16 @@ if (typeof Mozilla === 'undefined') {
         var $language = $('#page-language-select');
         var previousLanguage = $language.val();
         $language.on('change', function() {
+            var newLanguage = $language.val();
             window.dataLayer.push({
                 'event': 'change-language',
-                'languageSelected': $language.val(),
+                'languageSelected': newLanguage,
                 'previousLanguage': previousLanguage
             });
-            $('#lang_form').attr('action', window.location.hash || '#').submit();
+            // get path without locale
+            var urlpath = window.location.pathname.slice(1).split('/').slice(1).join('/');
+            var newpath =  '/' + newLanguage + '/' + urlpath + window.location.search;
```

Comment:
It might be nice to add a helper method (and associated test) for this URL parsing/rebuilding functionality. Maybe something like `Utils.switchLangauge(pathName, newLang, queryString)`. That function would then be pure and easily testable.